### ISSUE TITLE
text::DescriptorEncoder writes incorrect metric names

### ISSUE THREAD

**Issue by olix0r** [prometheus/client_rust#186]
I have a metric that is reported via the Collector API:

```rust
                let clock_ticks = stat.utime + stat.stime;
                let cpu =
                    prom::ConstCounter::new(Duration::from_millis(clock_ticks * mpt).as_secs_f64());
                let cpue = encoder.encode_descriptor(
                    "cpu",
                    "Total user and system CPU time spent in seconds",
                    Some(&prom::Unit::Seconds),
                    prom::metrics::MetricType::Counter,
                )?;
                cpu.encode(cpue)?;
```

When it is encoded, its name is encoded inconsistently:

```
# HELP process_cpu_seconds Total user and system CPU time spent in seconds
# TYPE process_cpu_seconds counter
# UNIT process_cpu_seconds seconds
process_cpu_seconds_total 14.29
```

That is, the `_total` suffix is not appended via [`encoding::text::DescriptorEncoder::encode_descriptor`](https://github.com/prometheus/client_rust/blob/23c31bdacdc2445b0b85e79684a6d2534330fc29/src/encoding/text.rs#L80-L127). This needs to be updated to have the same write_suffix behavior that MetricEncoder uses to add _total, _info, etc.

**Comment by olix0r** [comment]
I found [this reference](https://github.com/OpenObservability/OpenMetrics/blob/1386544931307dff279688f332890c31b6c5de36/specification/OpenMetrics.md#overall-structure) that indicates that this is correct. My problem is actually that [prometheus-parse](https://github.com/ccakes/prometheus-parse-rs) doesn't handle this properly.

**Comment by mxinden** [comment]
Sorry for the delay here. Your second comment is correct, i.e. this is the correct syntax.

Anyways, I appreciate the detailed report. Let me know in case there is anything I can help with @olix0r.
